hal/src/rtl872x/ble_hal.cpp

```
@@ -1953,17 +2023,13 @@ int BleGap::setPairingConfig(const hal_ble_pairing_config_t* config) {
     CHECK_TRUE(config, SYSTEM_ERROR_INVALID_ARGUMENT);
     CHECK_FALSE(connected(nullptr), SYSTEM_ERROR_INVALID_STATE);
     CHECK_FALSE(isScanning_, SYSTEM_ERROR_INVALID_STATE);
-    bool adv = isAdvertising();
-    bool ok = false;
-    SCOPE_GUARD({
-        if (ok && adv) {
-            startAdvertising();
-        }
-    });
     if (btStackStarted_) {
         CHECK(stop());
         CHECK(init());
     }
+    SCOPE_GUARD({
+        start();
```

This is no longer `CHECK()`'d. Could `start()` fail and the error be not returned to the caller of `setPairingConfig()`? Or is this not a concern?

---

If `start()` failed, then the stack is resided in an invalid state and the stack will try restarting again when next BLE API that depends on the stack is called. So it's fine to not check the resut. Also I didn't ever notice the failure of the `start()` so far.